openbb_platform/providers/coingecko/openbb_coingecko/models/crypto_price.py

```diff
@@ -0,0 +1,169 @@
+        default=True,
+        description="Include 24hr volume in the response.",
+    )
+    include_24hr_change: bool = Field(
+        default=True,
+        description="Include 24hr change in the response.",
+    )
+    include_last_updated_at: bool = Field(
+        default=True,
+        description="Include last updated timestamp in the response.",
+    )
+    precision: Optional[str] = Field(
+        default=None,
+        description=(
+            "The precision of the data. Use 'full' for full precision, "
+            "otherwise 2 decimals."
+        ),
+    )
+
+    @field_validator("vs_currency", mode="before", check_fields=False)
+    @classmethod
+    def validate_vs_currency(cls, v: str) -> str:
+        """Validate and normalize vs_currency."""
+        return v.lower().strip()
+
+    @field_validator("symbol", mode="before", check_fields=False)
+    @classmethod
+    def validate_symbol(cls, v: Union[str, List[str]]) -> str:
+        """Validate and normalize symbol(s)."""
+        if isinstance(v, str):
+            return validate_symbol(v)
+        return ",".join([validate_symbol(symbol) for symbol in v])
+
+
+class CoinGeckoCryptoPriceData(CryptoPriceData):
+    """CoinGecko Real-time Crypto Price Data."""
+
+
+class CoinGeckoCryptoPriceFetcher(
+    Fetcher[
+        CoinGeckoCryptoPriceQueryParams,
+        List[CoinGeckoCryptoPriceData],
+    ]
+):
+    """Transform the query, extract and transform the data from the CoinGecko endpoints."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> CoinGeckoCryptoPriceQueryParams:
+        """Transform the query parameters."""
+        return CoinGeckoCryptoPriceQueryParams(**params)
+
+    @staticmethod
+    async def aextract_data(
+        query: CoinGeckoCryptoPriceQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Return the raw data from the CoinGecko endpoint."""
+        api_key = credentials.get("coingecko_api_key") if credentials else None
+        
+        symbols = query.symbol.split(",")
+        
+        # Build parameters for the API call
+        params = {
+            "ids": ",".join(symbols),
+            "vs_currencies": query.vs_currency,
+            "include_market_cap": str(query.include_market_cap).lower(),
+            "include_24hr_vol": str(query.include_24hr_vol).lower(),
+            "include_24hr_change": str(query.include_24hr_change).lower(),
+            "include_last_updated_at": str(query.include_last_updated_at).lower(),
+        }
+        
+        if query.precision:
+            params["precision"] = query.precision
+        
+        endpoint = "simple/price"
+        data = make_request(endpoint, params, api_key)
+        
+        if not data or not isinstance(data, dict):
+            raise EmptyDataError("No price data found.")
+        
+        results = []
+        
+        for coin_id, price_data in data.items():
```

Comment:
Sections like this should be handled under "transform data", in a way that iterates over all the items only once.

Output from this function should be as close as possible to the raw data output from the API.